### PR TITLE
Feature : Comet scripted fields

### PR DESCRIPTION
**Pull request by elarib**
## Summary
This PR add the possibility to add scripted fields in Comet.
Scripted fields compute data on the fly from the data being ingested.

**PR Type: Feature**

**Status: Ready to review**

**Breaking change? No**

### Solution
Now we can add the field scripted which is the sql part that create the field.  

Example :
 - id: long
 - name: String

In the domain we can have: 

      - name: "id"
        type: "long"
      - name: "name"
        type: "string"
      - name: nameUpperCase
        type: "string"
        scripted: "upper(name)"

Now in the output, we're going to have also a new column 'nameUpperCase' which contain the name upperCase
 
There was a problem when writing scripted: "input_file_name()", because when we apply this script we sometimes, already lost the inputFileName, so to fix that, we always now add a new column (comet_input_file_name) that we can use instead of input_file_name().

Please check the UT for more info.

### How has this been tested?
Just Unit tests for now

### Remaining Todos
This should be done before merging this PR:
* [x] - UT updated accordingly
* [ ] - Run it in a cluster

## Contributor checklist:
Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/test/scala/com/ebiznext/comet/database/extractor/ExtractScriptGenSpec.scala [hunk 1]
```diff
@@ -7,28 +7,28 @@ import org.scalatest.matchers.should.Matchers
...
-        getClass.getResource("/sample/database/EXTRACT_TABLE.sql.mustache").getPath
-      ),
-      templateParams
-    )
-
-    templatePayload shouldBe File(
-      getClass.getResource("/sample/database/expected_script_payload.txt").getPath
-    ).lines.mkString("\n")
-  }
+//
```
Comment (hayssams) [inline src/test/scala/com/ebiznext/comet/database/extractor/ExtractScriptGenSpec.scala:10]: Why did you comment this UT

**Comment by hayssams** [comment]
@elarib Do you think you'll be able submit this one again soon.
We are all very excited to start using it.

**Comment by elarib** [comment]
> @elarib Do you think you'll be able submit this one again soon.
> We are all very excited to start using it.

Yes, i just did. Please check :)

File: src/test/resources/sample/simple-json-locations/locations.yml [hunk 2]
```diff
@@ -30,3 +30,17 @@ schemas:
...
+      - name: "name_upper_case"
+        type: "string"
+        scripted: "upper(name)"
+        array: false
+        required: true
+        privacy: "NONE"
+        metricType: "NONE"
+      - name: "source_file_name"
+        type: "string"
+        scripted: regexp_extract(comet_input_file_name, '.+\/(.+)$' , 1)
```
Comment (hayssams) [inline src/test/resources/sample/simple-json-locations/locations.yml:42]: It would be gr8 if we could rename scripted as script

**Pull request merged**